pkg/noderesourcetopology/cache/overreserve.go

```diff
@@ -70,27 +70,27 @@ func NewOverReserve(lister listerv1alpha1.NodeResourceTopologyLister, indexer No
 	return obj, nil
 }
 
-func (ov *OverReserve) GetCachedNRTCopy(nodeName string, pod *corev1.Pod) (*topologyv1alpha1.NodeResourceTopology, bool) {
+func (ov *OverReserve) GetCachedNRTCopy(nodeName string, pod *corev1.Pod) (*topologyv1alpha1.NodeResourceTopology, error) {
 	ov.lock.Lock()
 	defer ov.lock.Unlock()
 	if ov.nodesWithForeignPods.IsSet(nodeName) {
-		return nil, false
+		return nil, fmt.Errorf("foreign pods scheduled on node: %s", nodeName)
 	}
 
 	nrt := ov.nrts.GetNRTCopyByNodeName(nodeName)
 	if nrt == nil {
-		return nil, true
+		return nil, fmt.Errorf("cannot get NodeResourceTopology for node: %s", nodeName)
```

Comment:
This is a trap in the existing code and would have deserved a comment. For historical reasons, NOT having NRT info is a legit case which is meant to be handled as passthrough (no filter). We can revisit this behavior of course, but in another PR :)

Reply:
I suggest to turn the error into a log and `return nil, nil` here